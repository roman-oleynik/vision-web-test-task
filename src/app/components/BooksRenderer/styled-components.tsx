import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;

  @media (min-width: 576px) {
    & {
      flex-direction: row;
      justify-content: center;
      align-items: stretch;
      flex-wrap: wrap;
    }
  }
`;

export const BookContainer = styled.div`
  min-height: 40px;
  width: 300px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  padding: 10px;
  margin: 5px;
  border: 2px solid #202124;
  border-radius: 6px;
`;

export const BookTitle = styled.h2`
  font-family: 'Poppins', sans-serif;
  font-size: 24px;
  line-height: 60px;
  color: black;
`;

export const BookAuthor = styled.i`
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
  line-height: 30px;
  color: grey;
  text-align: center;
`;

export const Text = styled.p`
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
  line-height: 30px;
  color: grey;
  text-align: center;
`;