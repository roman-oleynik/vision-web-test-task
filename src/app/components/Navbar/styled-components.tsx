import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

export const Nav = styled.nav`
  height: 70px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 30px;
  background: #202124;
`;

export const LinksContainer = styled.div`
  width: 160px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Title = styled.h1`
  font-family: 'Poppins', sans-serif;
  font-size: 24px;
  color: lightseagreen;
`;

export const Link = styled(NavLink)`
  color: #fff;
  text-decoration: none;
  font-family: 'Poppins', sans-serif;
  font-size: 14px;
  line-height: 24px;
  border-bottom: 2px solid transparent;

  &.active {
    border-bottom: 2px solid tomato;
    color: tomato;
  }
`;