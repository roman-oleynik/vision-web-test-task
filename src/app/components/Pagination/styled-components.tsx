import styled from 'styled-components';

export const Container = styled.nav`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px;
`;


export const PageLink = styled.button`
  height: 40px;
  width: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 3px;
  color: #fff;
  background: tomato;
  text-decoration: none;
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
  border: 0;
`;
export const SwitchButton = styled(PageLink)`
  height: 30px;
  width: 30px;
  border-radius: 50%;

  &:disabled {
    background: grey;
  }
`;
export const SelectedPageLink = styled(PageLink)`
  background: lightseagreen;
`;