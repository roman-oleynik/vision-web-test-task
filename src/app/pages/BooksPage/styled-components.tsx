import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
`;

export const Title = styled.h1`
  font-family: 'Poppins', sans-serif;
  font-size: 32px;
  line-height: 60px;
  color: black;
  text-align: center;
`;

export const Text = styled.p`
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
  line-height: 30px;
  color: grey;
  text-align: center;
`;

export const Button = styled(Link)`
  height: 30px;
  width: 100px;
  background: tomato;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 20px;
  color: #fff;
  text-decoration: none;
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
`;