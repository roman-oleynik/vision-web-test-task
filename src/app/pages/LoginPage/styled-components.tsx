import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
`;

export const Title = styled.h1`
  font-family: 'Poppins', sans-serif;
  font-size: 32px;
  line-height: 60px;
  color: black;
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Label = styled.label`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  font-family: 'Poppins', sans-serif;
  padding 10px 0;
`;

export const Input = styled.input`
  width: 280px;
  height: 30px;
  font-size: 16px;
  color: black;
`;

export const Button = styled.button`
  height: 30px;
  width: 100px;
  background: tomato;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 20px;
  color: #fff;
  text-decoration: none;
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
  border: 0;
`;

export const ErrorText = styled.p`
  font-family: 'Poppins', sans-serif;
  font-size: 12px;
  color: tomato;
`;
