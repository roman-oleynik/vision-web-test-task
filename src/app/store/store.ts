import {rootReducer} from '../reducers/rootReducer';
import { createLogger } from 'redux-logger';
import {createStore, applyMiddleware} from 'redux';

const loggerMiddleware = createLogger();

export const store = createStore(
	rootReducer,
	applyMiddleware(loggerMiddleware),
);
