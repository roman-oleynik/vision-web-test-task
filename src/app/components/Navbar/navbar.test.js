import React from 'react';
import { create } from 'react-test-renderer';
import { Navbar } from '.';
import rrd, { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../store/store';
import { LinksContainer, Nav, Title, Link } from './styled-components';


describe("Navbar", () => {
    rrd.BrowserRouter = () => <div></div>;
    it("should match snapshot", () => {
        const component = create(
            <BrowserRouter>
                <Provider store={store}>
                    <Navbar />
                </Provider>
            </BrowserRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    test("The Home link navigates to '/'", () => {
        const component = create(
            <BrowserRouter>
                <Provider store={store}>
                    <Navbar />
                </Provider>
            </BrowserRouter>
        );
        const homeLink = component.root.findAllByType(Link)[0];
        expect(homeLink.props.to).toBe("/");
    });
    test("The Books link navigates to '/books'", () => {
        const component = create(
            <BrowserRouter>
                <Provider store={store}>
                    <Navbar />
                </Provider>
            </BrowserRouter>
        );
        const booksLink = component.root.findAllByType(Link)[1];
        expect(booksLink.props.to).toBe("/books");
    });
    test("The Profile link navigates to '/profile'", () => {
        const component = create(
            <BrowserRouter>
                <Provider store={store}>
                    <Navbar />
                </Provider>
            </BrowserRouter>
        );
        const profileLink = component.root.findAllByType(Link)[2];
        expect(profileLink.props.to).toBe("/profile");
    });
})