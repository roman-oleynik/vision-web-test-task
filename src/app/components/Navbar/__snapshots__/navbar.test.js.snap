// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match snapshot 1`] = `
<nav
  className="sc-bdfBwQ byPhtD"
>
  <h1
    className="sc-dlfnbm kdlcjy"
  >
    Bamazon
  </h1>
  <div
    className="sc-gsTCUz juqLjK"
  >
    <a
      aria-current="page"
      className="sc-hKgILt bTFXEO active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="sc-hKgILt bTFXEO"
      href="/books"
      onClick={[Function]}
    >
      Books
    </a>
    <a
      aria-current={null}
      className="sc-hKgILt bTFXEO"
      href="/profile"
      onClick={[Function]}
    >
      Profile
    </a>
  </div>
</nav>
`;
